docs/Advanced-Usage/Conda.rst

```diff
@@ -0,0 +1,224 @@
+this ::
+
+    bash-4.2$ conda update moto
+    Collecting package metadata (current_repodata.json): done
+    Solving environment: done
+
+    ## Package Plan ##
+
+      environment location: /opt/conda
+
+      added / updated specs:
+        - moto
+
+
+    The following NEW packages will be INSTALLED:
+
+      graphql-core       conda-forge/noarch::graphql-core-3.2.0-pyhd8ed1ab_0
+
+    The following packages will be UPDATED:
+
+      moto                                  2.2.19-pyhd8ed1ab_0 --> 3.1.0-pyhd8ed1ab_0
+
+    Proceed ([y]/n)?
+
+
+The addition of ``graphql-core`` makes sense because the `diff of moto's setup.py between
+2.2.19 and 3.1.0 <https://github.com/spulec/moto/compare/2.2.19...3.1.0#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7>`_
+shows it was clearly added as a new dependence.
+
+And this output tells us that latest version of ``moto`` available is 3.1.0.  Now, you might be tempted to
+hit ``<<Enter>>`` and move forward with your life.
+
+.. attention::
+
+    However, it is always a better idea to modify the version in ``machine-launch-script.sh`` so that:
+    #. you remember to commit and share the new version requirement.
+    #. you are providing a complete set of requirements for ``conda`` to solve.  There is a subtle difference between
+
+
+So, modify ``machine-launch-script.sh`` with the updated version of ``moto``, and run it.  If you'd like to see what
+``machine-launch-script.sh`` will do before actually making changes to your environment, feel free to give it the ``--dry-run``
+option, look at the output and then run again without ``--dry-run``.
+
+In this case, when you are finished, you can run ``conda list --revisions`` and you should see output
+like the following ::
+
+    bash-4.2$ conda list --revisions
+    2022-03-15 19:21:10  (rev 0)
+    +_libgcc_mutex-0.1 (conda-forge/linux-64)
+    +_openmp_mutex-4.5 (conda-forge/linux-64)
+    +_sysroot_linux-64_curr_repodata_hack-3 (conda-forge/noarch)
+    +alsa-lib-1.2.3 (conda-forge/linux-64)
+    +appdirs-1.4.4 (conda-forge/noarch)
+    +argcomplete-1.12.3 (conda-forge/noarch)
+
+     ...   many packages elided for this example ...
+
+    +xxhash-0.8.0 (conda-forge/linux-64)
+    +xz-5.2.5 (conda-forge/linux-64)
+    +yaml-0.2.5 (conda-forge/linux-64)
+    +zipp-3.7.0 (conda-forge/noarch)
+    +zlib-1.2.11 (conda-forge/linux-64)
+    +zstd-1.5.2 (conda-forge/linux-64)
+
+    2022-03-15 19:34:06  (rev 1)
+         moto  {2.2.19 (conda-forge/noarch) -> 3.1.0 (conda-forge/noarch)}
+
+This shows you that the first time ``machine-launch-script.sh`` was run, it created 'revision' 0 of the environment with
+many packages.  After updating the version of ``moto`` and rerunning, 'revision' 1 was created by updating the version
+of ``moto``.  At any time, you can revert your conda environment back to an older 'revision' using ``conda install -revision <n>``
```

Comment:
This is awesome!

Reply:
I wonder if it makes sense to "tag", in some capacity, the revision with the sha that produced it, but this is great.